src/identity-providers/add/DescriptorSettings.tsx

```diff
@@ -0,0 +1,428 @@
+            forLabel={t("nameIdPolicyFormat")}
+            forID="kc-nameIdPolicyFormat"
+          />
+        }
+        fieldId="kc-nameIdPolicyFormat"
+        helperTextInvalid={t("common:required")}
+      >
+        <Controller
+          name="config.nameIDPolicyFormat"
+          defaultValue={t("persistent")}
+          control={control}
+          render={({ onChange, value }) => (
+            <Select
+              toggleId="kc-nameIdPolicyFormat"
+              onToggle={() =>
+                setNamedPolicyDropdownOpen(!namedPolicyDropdownOpen)
+              }
+              isOpen={namedPolicyDropdownOpen}
+              onSelect={(_, value) => {
+                onChange(value as string);
+                setNamedPolicyDropdownOpen(false);
+              }}
+              selections={value}
+              variant={SelectVariant.single}
+            >
+              <SelectOption
+                key={0}
+                data-testid="persistent-option"
+                value={t("persistent")}
+                isPlaceholder
+              />
+              <SelectOption
+                key={1}
+                data-testid="transient-option"
+                value={t("transient")}
+              />
+              <SelectOption
+                key={2}
+                data-testid="email-option"
+                value={t("email")}
+              />
+              <SelectOption
+                key={3}
+                data-testid="kerberos-option"
+                value={t("kerberos")}
+              />
+              <SelectOption
+                key={4}
+                data-testid="x509-option"
+                value={t("x509")}
+              />
+              <SelectOption
+                key={5}
+                data-testid="windowsDomainQN-option"
+                value={t("windowsDomainQN")}
+              />
+              <SelectOption
+                key={6}
+                data-testid="unspecified-option"
+                value={t("unspecified")}
+              />
+            </Select>
+          )}
+        ></Controller>
+      </FormGroup>
+
+      <FormGroup
+        label={t("principalType")}
+        labelIcon={
+          <HelpItem
+            helpText={th("principalType")}
+            forLabel={t("principalType")}
+            forID="kc-principalType"
+          />
+        }
+        fieldId="kc-principalType"
+        helperTextInvalid={t("common:required")}
+      >
+        <Controller
+          name="config.principalType"
+          defaultValue={t("subjectNameId")}
+          control={control}
+          render={({ onChange, value }) => (
+            <Select
+              toggleId="kc-principalType"
+              onToggle={() =>
+                setPrincipalTypeDropdownOpen(!principalTypeDropdownOpen)
+              }
+              isOpen={principalTypeDropdownOpen}
+              onSelect={(_, value) => {
+                onChange(value as string);
```

Comment:
There is no need to cast this type as `SelectOptionObject` implements `toString()`.
```suggestion
                onChange(value.toString());
```